component/src/main/java/org/wso2/carbon/identity/authenticator/pinterest/PinterestAuthenticator.java

```diff
@@ -0,0 +1,396 @@
+
+	/**
+	 * check weather user can process or not.
+	 *
+	 * @param request the request
+	 * @return true or false
+	 */
+	@Override
+	public boolean canHandle(HttpServletRequest request) {
+		if (log.isDebugEnabled()) {
+			log.debug("Inside PinterestOAuth2Authenticator canHandle method and checking whether the code and state " +
+			          "exist");
+		}
+		return request.getParameter(PinterestAuthenticatorConstants.OAUTH2_GRANT_TYPE_CODE) != null &&
+		       request.getParameter(PinterestAuthenticatorConstants.OAUTH2_PARAM_STATE) != null;
+	}
+
+	/**
+	 * Get Pinterest authorization endpoint.
+	 */
+	@Override
+	protected String getAuthorizationServerEndpoint(Map<String, String> authenticatorProperties) {
+		return PinterestAuthenticatorConstants.PINTEREST_OAUTH_ENDPOINT;
+	}
+
+	/**
+	 * Get Pinterest token endpoint.
+	 */
+	@Override
+	protected String getTokenEndpoint(Map<String, String> authenticatorProperties) {
+		return PinterestAuthenticatorConstants.PINTEREST_TOKEN_ENDPOINT;
+	}
+
+	/**
+	 * Get Pinterest user info endpoint.
+	 */
+	@Override
+	protected String getUserInfoEndpoint(OAuthClientResponse token, Map<String, String> authenticatorProperties) {
+		return PinterestAuthenticatorConstants.PINTEREST_USERINFO_ENDPOINT;
+	}
+
+	/**
+	 * Check ID token in Pinterest OAuth.
+	 */
+	@Override
+	protected boolean requiredIDToken(Map<String, String> authenticatorProperties) {
+		return false;
+	}
+
+	/**
+	 * Get the friendly name of the Authenticator.
+	 */
+	@Override
+	public String getFriendlyName() {
+		return PinterestAuthenticatorConstants.PINTEREST_CONNECTOR_FRIENDLY_NAME;
+	}
+
+	/**
+	 * Get the name of the Authenticator.
+	 */
+	@Override
+	public String getName() {
+		return PinterestAuthenticatorConstants.PINTEREST_CONNECTOR_NAME;
+	}
+
+	/**
+	 * Get Configuration Properties.
+	 */
+	@Override
+	public List<Property> getConfigurationProperties() {
+		List<Property> configProperties = new ArrayList<>();
+
+		Property clientId = new Property();
+		clientId.setName(OIDCAuthenticatorConstants.CLIENT_ID);
+		clientId.setDisplayName(PinterestAuthenticatorConstants.CLIENT_ID);
+		clientId.setRequired(true);
+		clientId.setDescription("Enter Pinterest IDP client identifier value");
+		clientId.setDisplayOrder(0);
+		configProperties.add(clientId);
+
+		Property clientSecret = new Property();
+		clientSecret.setName(OIDCAuthenticatorConstants.CLIENT_SECRET);
+		clientSecret.setDisplayName(PinterestAuthenticatorConstants.CLIENT_SECRET);
+		clientSecret.setRequired(true);
+		clientSecret.setConfidential(true);
+		clientSecret.setDescription("Enter Pinterest IDP client secret value");
+		clientSecret.setDisplayOrder(1);
+		configProperties.add(clientSecret);
+
+		Property callbackUrl = new Property();
+		callbackUrl.setDisplayName("Callback URL");
+		callbackUrl.setName(PinterestAuthenticatorConstants.CALLBACK_URL);
+		callbackUrl.setDescription("Enter value corresponding to callback url.");
```

Comment:
Fixed in https://github.com/wso2-extensions/identity-outbound-auth-pinterest/pull/1/commits/2d20b214a60d365dbfdad4f646691e271a536769